clinica/iotools/converters/aibl_to_bids/utils/clinical.py

```diff
@@ -168,263 +208,119 @@ def create_sessions_tsv_file(
     """
     import glob
 
-    from clinica.iotools.bids_utils import StudyName, bids_id_factory
+    from clinica.iotools.bids_utils import (
+        StudyName,
+        bids_id_factory,
+        get_bids_sess_list,
+        get_bids_subjs_list,
+    )
 
+    study = StudyName.AIBL.value
     specifications = _load_specifications(
         clinical_specifications_folder, "sessions.tsv"
-    )
-    sessions_fields = specifications[StudyName.AIBL.value]
-    field_location = specifications[f"{StudyName.AIBL.value} location"]
-    sessions_fields_bids = specifications["BIDS CLINICA"]
-    fields_dataset = []
-    fields_bids = []
-
-    for i in range(0, len(sessions_fields)):
-        if not pd.isnull(sessions_fields[i]):
-            fields_bids.append(sessions_fields_bids[i])
-            fields_dataset.append(sessions_fields[i])
+    )[["BIDS CLINICA", f"{study} location", study]].dropna()
 
-    files_to_read: List[str] = []
-    sessions_fields_to_read: List[str] = []
-    for i in range(0, len(sessions_fields)):
-        # If the i-th field is available
-        if not pd.isnull(sessions_fields[i]):
-            # Load the file
-            file_to_read_path = clinical_data_dir / field_location[i]
-            files_to_read.append(glob.glob(str(file_to_read_path))[0])
-            sessions_fields_to_read.append(sessions_fields[i])
-
-    rids = pd.read_csv(files_to_read[0], dtype={"text": str}, low_memory=False).RID
-    unique_rids = list(set(rids))
-    for rid in unique_rids:
-        for file in files_to_read:
-            df = pd.read_csv(file, dtype={"text": str})
-            if len(df.columns) == 1:
-                df = pd.read_csv(file, sep=";", low_memory=False)
-
-            visit_code = df.loc[(df["RID"] == rid), "VISCODE"]
-
-            for field in sessions_fields_to_read:
-                if field in list(df.columns.values) and field == "MMSCORE":
```

Comment:
I don't see these hardcoded values in your code. 
I'm probably missing something, but how do you handle these in your version ?

Reply:
To explain what my code does quickly : 
- I read the specifications file where it is indicated for each metadata of interest : name in the study, name for BIDS, location in csv files
--> I don't really need to hard code anything for metadata extraction since I loop over what is in the spec file. The process for "extraction" is the same, which you can find in the _format_metadata_for_rid function.
- If needed I perform some mapping which is similar to what is done here but I use the BIDS name instead. For ex my "diagnosis" mapping corresponds to what was done for field "DXCURREN"